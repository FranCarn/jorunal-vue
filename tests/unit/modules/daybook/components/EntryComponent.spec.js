import { shallowMount } from "@vue/test-utils";
import EntryComponent from "@/modules/daybook/components/EntryComponent";

describe("tests on entry component", () => {
  const mockRouter = {
    push: jest.fn(),
  };
  const wrapper = shallowMount(EntryComponent, {
    props: {
      entry: {
        date: 1627077227978,
        text: "Hi Jest",
        id: "-NraVTzAqgA-CtJSWRtq",
      },
    },
    global: { mocks: { $router: mockRouter } },
  });

  test("should match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should redirect when user click entry-container div", () => {
    const div = wrapper.find(".entry-container");
    div.trigger("click");

    expect(mockRouter.push).toBeCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: "-NraVTzAqgA-CtJSWRtq" },
    });
  });

  test("should check computed properties", () => {
    const { day, month, yearDay } = wrapper.vm;
    expect(day).toBe(23);
    expect(month).toBe("July");
    expect(yearDay).toBe("2021, Saturday");
  });
});
