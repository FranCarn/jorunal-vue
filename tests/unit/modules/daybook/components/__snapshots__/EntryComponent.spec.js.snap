// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on entry component should match with snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">July</span><span class="mx-2 fw-light">2021, Saturday</span></div>
  <div class="entry-description">Hi Jest</div>
</div>
`;
